src/main/java/com/github/fridujo/rabbitmq/mock/MockPolicy.java

```diff
@@ -0,0 +1,77 @@
+package com.github.fridujo.rabbitmq.mock;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Singular;
+import lombok.ToString;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import static com.github.fridujo.rabbitmq.mock.MockPolicy.ApplyTo.ALL;
+import static com.github.fridujo.rabbitmq.mock.tool.ParameterMarshaller.getParameterAsExchangePointer;
+import static com.github.fridujo.rabbitmq.mock.tool.ParameterMarshaller.getParameterAsString;
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+
+@Getter
+@ToString
+public class MockPolicy {
+    public static final String ALTERNATE_EXCHANGE = "alternate-exchange";
+    public static final String DEAD_LETTER_EXCHANGE = "dead-letter-exchange";
+    public static final String DEAD_LETTER_ROUTING_KEY = "dead-letter-routing-key";
+
+    private String name;
+    private String pattern;
+    private Integer priority;
+    private Map<String, Object> definitions;
+    private ApplyTo applyTo;
+
+    @Builder(toBuilder = true)
+    public MockPolicy(@NonNull String name, @NonNull String pattern, @NonNull @Singular Map<String, Object> definitions,
+                      Integer priority, ApplyTo applyTo) {
+        this.name = name;
+        this.pattern = pattern;
+        this.definitions = definitions;
+        this.priority = priority == null ? 0 : priority;
+        this.applyTo = applyTo == null ? ALL : applyTo;
+    }
+
+    public Optional<ReceiverPointer> getAlternateExchange() {
+        return getParameterAsExchangePointer.apply(ALTERNATE_EXCHANGE, definitions);
+    }
+
+    public Optional<ReceiverPointer> getDeadLetterExchange() {
+        return getParameterAsExchangePointer.apply(DEAD_LETTER_EXCHANGE, definitions);
+    }
+
+    public Optional<String> getDeadLetterRoutingKey() {
+        return getParameterAsString.apply(DEAD_LETTER_ROUTING_KEY, definitions);
+    }
+
+    static final Comparator<MockPolicy> comparator = Comparator.comparing(MockPolicy::getPriority).reversed();
```

Comment:
Codacy found an issue: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/ledoyen/rabbitmq-mock/pullRequest?prid=5817489)